src/lib/core/style/add-focus-classes.spec.ts

```diff
@@ -0,0 +1,112 @@
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {Component} from '@angular/core';
+import {StyleModule} from './index';
+import {By} from '@angular/platform-browser';
+
+
+describe('cdkFocusClasses', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [StyleModule],
+      declarations: [
+        ButtonWithFocusClasses,
+      ],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  describe('cdkFocusClasses', () => {
+    let fixture: ComponentFixture<ButtonWithFocusClasses>;
+    let buttonElement: HTMLElement;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(ButtonWithFocusClasses);
+      fixture.detectChanges();
+
+      buttonElement = fixture.debugElement.query(By.css('button')).nativeElement;
+    });
+
+    it('should initially not be focused', () => {
+      expect(buttonElement.classList.length).toBe(0, 'button should not have focus classes');
+    });
+
+    it('should detect focus via keyboard', async(() => {
+      // Simulate focus via keyboard.
+      dispatchKeydownEvent(document, 9 /* tab */);
```

Comment:
You should be able to use the `TAB` constant from [here](https://github.com/angular/material2/blob/master/src/lib/core/keyboard/keycodes.ts#L20)